test/e2e/autoscale_test.go

```diff
@@ -202,6 +203,17 @@ func assertScaleUp(ctx *testContext) {
 	if err != nil {
 		ctx.t.Fatalf("Unable to observe the Deployment named %s scaling up. %s", ctx.deploymentName, err)
 	}
+
+	dep, err := ctx.clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace).Get(ctx.deploymentName, metav1.GetOptions{})
```

Comment:
I think this all is done on line 196, isn't it?

Reply:
You are right.  That check is already in place.   I added validation of revision at setup() instead.